### ISSUE TITLE
Freezes when opening large remote projects

### ISSUE THREAD

**Issue by kelmes** [hlissner/doom-emacs#5360]
### What did you expect to happen?

Projectile should (I think?) ignore .gitignore'd files in a git repo.

### What actually happened?

Opening a remote project with ~600,000 untracked/ignored files freezes emacs, then after a few minutes it crashes.

### Describe your attempts to resolve the issue

`(projectile-files-via-ext-command "/ssh:host:path/" "git ls-files -zco --exclude-standard")` correctly ignores everything and only takes a second.

`(print projectile-git-command)` shows `git ls-files -zco --exclude-standard`

after `(setq projectile-enable-caching nil)` we get `Switched to 'path'` instead of the projectile cache line, then emacs freezes anyway.

### Steps to reproduce

Add a remote project with `spc+p+a` , this should have a large number of untracked .gitignore'd files

Switch to this project for the first time w/ `spc+p+p`

Emacs shows "Projectile is initializing cache for /ssh:host:/project_path/`, then freezes and eventually crashes.

### System Information

https://pastebin.com/e2M4qbAw

**Comment by danilevy1212** [comment]
Hi, I've had this issue in the past. I haven't "fixed" it but made it somewhat better with these changes, first on tramp settings:

```elisp
(after! tramp
  (setq remote-file-name-inhibit-cache 60 ;; Less frequent cache write on remote host
        tramp-completion-reread-directory-timeout 60 ;; Less frequent cache write on remote host
        tramp-verbose 1                              ;; Remove verbosity
        vc-ignore-dir-regexp (format "%s\\|%s\\|%s"
                                           vc-ignore-dir-regexp
                                           tramp-file-name-regexp
                                           "[/\\\\]node_modules"))) ;; Disable remote VC
```
And on projectile:

```
(defadvice! dan/disable-projectile-project-root-on-remote (org-fun &rest args)
  "Disable search of the projectile root name on remote hosts."
  :around projectile-project-root
  (if (file-remote-p default-directory)
      (setq-local projectile-mode-line "Projectile")
    (apply 'org-fun args)))
```

These two blocks make significant change in speed when opening remote projects, although it's still is some-what slow. Hope it helps.

**Comment by hlissner** [comment]
As of 0b5243c this issue should be resolved. I've worked a few of @danilevy1212's suggestions into our defaults, though the advice would not work for a couple reasons:

- The syntax is incorrect. `projectile-project-root` should be quoted.
- Projectile does not use `projectile-mode-line` (at least, not in the latest version Doom has pinned), but I've adapted a new version of it in 0b5243c which should.

I've also changed projectile to fall back to its VCS shell commands when collecting project information in remote buffers, which should help OP's case, specifically. Altogether, TRAMP performance should be better.

Let me know if the issue resurfaces and I'll reopen it. Thanks for bringing it to my attention!
